### PR TITLE
More thoroughly explain concepts that might confuse beginners in Tutorial

### PR DESCRIPTION
**Pull request by martijnbastiaan**

### CODE REVIEW

File: clash-prelude/src/Clash/Tutorial.hs [hunk 1]
```diff
@@ -407,7 +439,13 @@ examine the 'register' function by taking a look at the first 4 samples of the
 [0,0,8,8]
 
 Where we see that the initial value of the signal is the specified 0 value,
-followed by 8's.
+followed by 8's. You might be surprised to see /two/ zeros instead of just a
+single zero. What happens is that Clash emulates what happens /before/ the
```
Comment (christiaanb) [inline clash-prelude/src/Clash/Tutorial.hs]: I'm not a big fan of the word 'emulate'.

Perhaps: "What happens is that in Clash you get to see the output of the circuit /before/ the clock becomes actives. In other words, in Clash you get to describe the powerup values of registers too."

File: clash-prelude/src/Clash/Tutorial.hs [hunk 2]
```diff
@@ -407,7 +439,13 @@ examine the 'register' function by taking a look at the first 4 samples of the
...
+clock becomes active. In other words, Clash emulates the powerup values of
+registers too. Whether this is a defined or undefined value depends on your
```
Comment (christiaanb) [inline clash-prelude/src/Clash/Tutorial.hs]:
```suggestion
registers too. Whether this is a defined or unknown value depends on your
```

File: clash-prelude/src/Clash/Tutorial.hs [hunk 3]
```diff
@@ -407,7 +439,13 @@ examine the 'register' function by taking a look at the first 4 samples of the
...
+synthesis target, and can be configured by using a different synthesis
```
Comment (christiaanb) [inline clash-prelude/src/Clash/Tutorial.hs]:
```suggestion
hardware target, and can be configured by using a different synthesis
```

File: clash-prelude/src/Clash/Tutorial.hs [hunk 4]
```diff
@@ -354,22 +354,53 @@ We can now finally start describing the logic of our circuit, starting with just
 the multiplication and addition:
 
 @
-ma acc (x,y) = acc + x * y
+ma acc (x, y) = acc + x * y
 @
 
+The circuit we just wrote is a combinational circuit: no registers are inserted
+(all registers are inserted explicitly in Clash, as we'll later see). We usually
```
Comment (christiaanb) [inline clash-prelude/src/Clash/Tutorial.hs]:
```suggestion
(you describe explicitly where Clash will insert registers, as we'll later see). We usually
```

File: clash-prelude/src/Clash/Tutorial.hs [hunk 5]
```diff
@@ -354,22 +354,53 @@ We can now finally start describing the logic of our circuit, starting with just
...
+refer to circuits as /functions/, similar to programming languages such as C,
+Python, or Haskell. In this case, the function we just defined is called @ma@.
+Its first argument is @acc@, its second is @(x, y)@ - a composite type called a
+tuple. This component is "unpacked", and its first element is called @x@, its
+second @y@. Everything to the right of the equals symbol is @ma@'s
+implementation.
```
Comment (christiaanb) [inline clash-prelude/src/Clash/Tutorial.hs]:
```suggestion
result.
```

File: clash-prelude/src/Clash/Tutorial.hs [hunk 6]
```diff
@@ -354,22 +354,53 @@ We can now finally start describing the logic of our circuit, starting with just
...
+ * __@a@__, @ma@'s first argument is of type @a@
+
+ * __@(a, a)@__, @ma@'s second argument is of type @(a, a)@
+
+ * __@a@__, @ma@'s result is of type @a@
+
+Note that @ma@ therefore works on multiple types! The only condition we
+imposed is that @a@ should be a @'Num'@ber type. In Clash this means it should
+support the operations @'Prelude.+'@, @'Prelude.-'@, @'Prelude.*'@, and some
+others. Indeed, this is why Clash added the constraint in the first place: the
```
Comment (christiaanb) [inline clash-prelude/src/Clash/Tutorial.hs]:
```suggestion
others. Indeed, this is why Clash adds the constraint in the first place: the
```

**Comment by martijnbastiaan** [comment]
Thanks @christiaanb !

**Pull request merged**